pyximport/pyximport.py

```diff
@@ -185,12 +185,12 @@ def build_module(name, pyxfilename, pyxbuild_dir=None, inplace=False, language_l
     from . import pyxbuild
     olddir = os.getcwd()
     common = ''
-    if pyxbuild_dir:
+    if pyxbuild_dir and sys.platform == 'win32':
         # Windows concatenates the pyxbuild_dir to the pyxfilename when
         # compiling, and then complains that the filename is too long
         common = os.path.commonprefix([pyxbuild_dir, pyxfilename])
     if len(common) > 30:
-        pyxfilename = os.path.relpath(pyxfilename)
+        pyxfilename = os.path.relpath(pyxfilename, common)
         pyxbuild_dir = os.path.relpath(pyxbuild_dir)
         os.chdir(common)
```

Comment:
This is a bug fix, since the `chdir` should have preceded the shortening of `pyxfilename` and `pyxbuild_dir`:
```
os.chdir(common)
pyxfilename = os.path.relpath(pyxfilename)
pyxbuild_dir = os.path.relpath(pyxbuild_dir)
```

Shouldn't the fix in the PR also apply to `pyxbuild_dir`?:
```
pyxbuild_dir = os.path.relpath(pyxbuild_dir, common)
```

Reply:
> `chdir` should have preceded the shortening of `pyxfilename` and `pyxbuild_dir`

Is there a guarantee that `pyxfilename` is absolute? If not, if you chdir first, your path to `pyxfilename` won't be correct any more.

> Shouldn't the fix in the PR also apply to pyxbuild_dir?:

Oh, yeah, probably